tests/settings/ConfigurationGatheringTest.py

```diff
@@ -281,3 +281,17 @@ def test_no_config(self):
                     ['--no-config', '--find-config'],
                     self.log_printer)
                 self.assertEqual(cm.exception.code, 2)
+
+    def test_section_inheritance(self):
+        current_dir = os.path.abspath(os.path.dirname(__file__))
+        test_dir = os.path.join(current_dir, 'section_manager_test_files')
+
+        with change_directory(test_dir):
+            sections, _, _, _ = gather_configuration(
+                lambda *args: True,
+                self.log_printer,
+                arg_list=['-c', 'inherit_coafile'])
+            self.assertEqual(sections['all.python'].defaults, sections['all'])
+            self.assertEqual(sections['all.c']['config'],
+                             sections['default']['config'])
+            self.assertEqual(sections['java.test'].defaults, sections['java'])
```

Comment:
Maybe test if `all.python[max_line_length] == 80`? After all that's the whole point of inheritence IIUC.